docs/madr/decisions/072-metrics-migration.md

```diff
@@ -0,0 +1,115 @@
+
+## Considered Options
+
+* Option 1 - Rename resources based on KRI without introducing a breaking change, and introduce a feature flag to enable the new resource naming.
+
+## Decision Outcome
+
+* Option 1 - ename resources based on KRI without introducing a breaking change, and introduce a feature flag to enable the new resource naming.
+
+## Pros and Cons of the Options
+
+### Option 1 - ename resources based on KRI without introducing a breaking change, and introduce a feature flag to enable the new resource naming.
+
+We currently have two modes to consider:
+
+* `kuma.io/service`
+* `MeshService`, `MeshExternalService`, and `MeshMultiZoneService`
+
+We will not change anything for `kuma.io/service`, since it is not a real resource.
+
+During this task, we should split the work into two main parts:
+* Rename places that are not exposed to or used by clients, so that we can adopt the new model by default internally.
+* Introduce a feature flag to switch all resource names and metrics to the KRI-based naming for users.
+
+#### Rename places that are not exposed to or used by clients, so that we can adopt the new model by default internally.
+
+We can distinguish two types of clients:
+
+* Internal — GUI
+* External — Dashboards, Prometheus
+
+On the Dataplane view, we show the outbounds and inbounds.
+We retrieve this information from the stats and later reference the configuration based on the name of the stats — either the cluster name or the listener name.
+Changing either the resource name or the statistic name might impact this functionality.
+We need to coordinate changes with the UI team to ensure we support both situations.
+
+What can we do?
+
+>[!NOTE]
+> Format of resource names is defined in [MADR](070-resource-identifier.md)
+
+* `Dataplane`
+Cluster:
+```
+  name: kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport
+  alt_stats_name: mesh-1_backend_kuma-demo_us-east2-_msvc_9090
+```
+Listener:
+```
+  name: kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport
+  stat_prefix: mesh-1_backend_kuma-demo_us-east2-_msvc_9090
+```
+The GUI should be updated to support looking up resources not just by name, but also by `alt_stats_name` (for Clusters) or `stat_prefix` (for Listeners).
+This way, we avoid breaking the stats view for metrics even when naming changes.
+
+* `ZoneIngress` and `ZoneEgress`
+In these cases, we should be able to rename the fields without any impact, since they are not exposed to external consumers.
+Cluster:
+```
+  name: kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport
+```
+Listener:
+```
+  name: kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport
+```
+
+#### Introduce a feature flag to switch all resource names and metrics to the KRI-based naming for users.
+
+In `2.11.x`, we want to introduce a flag that allows switching to the new format, along with a note that starting from `2.13.x`, this will become the default behavior.
```

Comment:
I'm not sure we should put this note yet. We'll say it will be default in the future but we probably shouldn't be specific wrt when